src/App.js

```diff
@@ -1,63 +1,124 @@
-      }
-    }, [todoList, isLoading]); // execute/call the callback function(effect) only when the 'todoList' state variable changes by specifying 'todoList' as the dependency array
-
+    try {
+      const response = await fetch(url, options);
 
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
+      // Parse the response JSON
+      const data = await response.json();
 
-    function removeTodo(id) {
-      // Filter the todoList array to exclude the todo item with the specified id
-      const updatedTodoList = todoList.filter((todo) => todo.id !== id);
+      // Transform API data.records into todo objects
+      const todos = data.records.map((record) => ({
+        id: record.id,
+        title: record.fields.title,
+      }));
 
-      // Update the todoList state with the new array of todos
-      setTodoList(updatedTodoList);
+      // Update todoList and isLoading states
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(`Fetch error: ${error.message}`);
     }
+  }
 
+  // Fetch data on initial render
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  // Handle adding a new todo
+  async function handleAddTodo(newTodo) {
+    try {
+      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          records: [{ fields: { title: newTodo.title } }],
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
-    // Handle adding a new todo
-    function addTodo(newTodo) {
-      setTodoList([...todoList, newTodo]); // Update todoList state by adding the new todo
+      const data = await response.json();
+      const confirmedTodo = data.records[0];
+
+      // Update the todoList state by adding the new todo
+      setTodoList([...todoList, { id: confirmedTodo.id, title: confirmedTodo.fields.title }]);
+    } catch (error) {
+      console.error('Error adding todo:', error.message);
     }
+  }
 
-    return (
-      <>
-        <AddTodoForm onAddTodo={addTodo} />
-        {isLoading ? <p>Loading...</p> : <TodoList todoList={todoList} onRemoveTodo={removeTodo} />}
-        
+  // Define the removeTodo function to remove a todo item
+  function removeTodo(id) {
+    // Filter the todoList array to exclude the todo item with the specified id
+    const updatedTodoList = todoList.filter((todo) => todo.id !== id);
 
-      </>
-    );
+    // Update the todoList state with the new array of todos
+    setTodoList(updatedTodoList);
   }
 
+
+  return (
+    <BrowserRouter>
```

Comment:
For future reference I find it easier to review PRs that depend on other open PRs to point to the base. This way only the lesson 1.9 changes would show up on this PR. Here's docs on how to do this: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

And to be explicit I'm suggesting changing the base branch of this PR to lesson_1_8 so that the airtable changes aren't shown in this PR.